src/main/java/baseball/model/BallNumbers.java

```diff
@@ -0,0 +1,31 @@
+package baseball.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BallNumbers {
+    private final List<Ball> ballList;
+
+    public BallNumbers(List<Ball> ballList) {
+        this.ballList = new ArrayList<>(ballList);
+    }
+
+    public Result compare(BallNumbers userBallNumbers) {
+        int strike = 0;
+        int ball = 0;
+
+        List<Ball> userBallList = userBallNumbers.ballList;
+        for (int i = 0; i < ballList.size(); i++) {
```

Comment:
```suggestion
        for (int ballOrder; ballOrder < ballList.size(); ballOrder++) {
```
이런 식으로 i 대신 의미가 명확한 변수명을 사용하는 것도 좋을 것 같습니다!